solidity_contracts/src/Kakarot/Coinbase.sol

```diff
@@ -1,23 +1,32 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.8.27;
 
-import {DualVmToken} from "../CairoPrecompiles/DualVmToken.sol";
 import {Ownable2Step, Ownable} from "@openzeppelin-contracts/access/Ownable2Step.sol";
+import {CairoLib} from "kakarot-lib/CairoLib.sol";
 
 contract Coinbase is Ownable2Step {
-    /// @dev The EVM address of the DualVmToken for Kakarot ETH.
-    DualVmToken public immutable kakarotEth;
+    using CairoLib for uint256;
+
+    uint256 public immutable nativeTokenStarknetAddress;
 
     /// Constructor sets the owner of the contract
-    constructor(address _kakarotEth) Ownable(msg.sender) {
-        kakarotEth = DualVmToken(_kakarotEth);
+    constructor(uint256 _nativeTokenStarknetAddress) Ownable(msg.sender) {
```

Comment:
The owner will be the déployer of the contract 
Do we want to keep this ?
Will this be the kkrt-labs multi sig or only the déployer of contracts (which I assume are different)